clang/lib/APINotes/APINotesManager.cpp

```diff
@@ -206,13 +204,13 @@ const DirectoryEntry *APINotesManager::loadFrameworkAPINotes(
     llvm::sys::path::append(Path, "PrivateHeaders");
 
   // Try to access the header directory.
-  auto HeaderDir = FileMgr.getDirectory(Path);
+  auto HeaderDir = FileMgr.getDirectoryRef(Path);
```

Comment:
This returns `Expected<DirectoryEntryRef>`; unlike `ErrorOr` you can't just discard error values, so this would abort if there is any error.  I think you want `getOptionalDirectoryRef`.  Same for other calls below.

Reply:
Great catch, thanks. Fixed in #7076.